metric/registry.go

```diff
@@ -53,6 +56,8 @@ func (r *Registry) AddInt64Gauge(name, description string, unit metricdata.Unit,
 }
 
 func (r *Registry) initGauge(g *gauge, labelKeys []string, name string, description string, unit metricdata.Unit) *gauge {
+	r.mu.Lock()
+	defer r.mu.Unlock()
 	existing, ok := r.gauges[name]
 	if ok {
```

Comment:
Instead of log.Panic probably we should return an error. you can file an issue for this.